packages/next/telemetry/events/build.ts

```diff
@@ -19,13 +19,30 @@ type EventBuildOptimized = {
   totalPageCount: number
   staticPageCount: number
   ssrPageCount: number
+  hasDunderPages: boolean
+  hasTestPages: boolean
 }
 
+const DUNDER_PAGES = /^[\\/]__generated__[\\/]/
+const TEST_PAGES = /^[\\/]__(tests|mocks)__[\\/]/
+const TEST_FILE = /\.(spec|test)\.[jt]sx?$/
```

Comment:
Should this take in account pageExtensions?
(e.g. `new RegExp(`\\.(?:${pageExtensions.join('|')})$`)`)